steward/src/cork.rs

```
@@ -84,11 +99,139 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
+        let client = SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id));
+
+        let cellar_address = request.cellar_id.clone();
+
+        if cellars::validate_cellar_id(&cellar_address).is_err() {
+            return Err(tonic::Status::invalid_argument(
+                "Error, can't validate Cellar ID",
+            ));
+        }
+
+        let address: Address = cellar_address
+            .parse()
+            .expect("could not parse cellar address");
+
+        let function_call = match contract_call_data {
+            AaveV2Stablecoin(call) => {
+                let contract = AaveV2StablecoinCellar::new(address, Arc::new(client));
+                match call.function.unwrap() {
+                    Function::EnterPosition(_) => contract_call(contract.enter_position()).await,
+                    Function::SetDepositLimit(params) => {
+                        contract_call(
+                            contract.set_deposit_limit(string_to_u256(params.limit).unwrap()),
+                        )
+                        .await
+                    }
+                    Function::Reinvest(params) => {
+                        contract_call(
+                            contract.reinvest(string_to_u256(params.min_assets_out).unwrap()),
+                        )
+                        .await
+                    }
+                    Function::Rebalance(params) => {
+                        let results: Vec<Result<H160, &String>> = params
+                            .route
+                            .iter()
+                            .map(|addr| match addr.parse::<H160>() {
+                                Ok(addr) => Ok(addr),
+                                Err(_) => Err(addr),
+                            })
+                            .collect();
+
+                        aave_v2_stablecoin::validate_route(results.clone()).unwrap_or_else(|err| {
+                            println!("{}", err);
```

If this fails we wouldn't want to proceed with the rest of the function and should return an error.